### PR TITLE
Add multi-ldap documentation

### PR DESCRIPTION
**Pull request by fviale**
Also, improved existing ldap documentation

### CODE REVIEW

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 1]
```diff
@@ -1291,15 +1291,17 @@ To be able to edit the default logo displayed in ProActive portals, you have to
 
 == User Authentication
 
-In order to use ProActive Scheduler every user must have an account. It supports three methods for authentication:
+In order to log into ProActive server, every user must have a registered account.
```
Comment (nebilbenmabrouk) [inline src/docs/admin/ProActiveAdminGuide.adoc:1294]: In order to log into ProActive **portals** (or simply **ProActive**)

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 2]
```diff
@@ -1291,15 +1291,17 @@ To be able to edit the default logo displayed in ProActive portals, you have to
...
-* *PAM*
+* <<_file,*File-based*>>
+* <<_ldap,*LDAP*>>
+* <<_ldap_with_multiple_domains,*LDAP with Multiple Domains*>>
+* <<_pam,*PAM*>>
 
 === Select authentication method
 
-By default the ProActive Scheduler is configured to use *file based* authentication and has some default accounts ('demo/demo', 'admin/admin') that
+By default, the ProActive server is configured to use *file based* authentication and has some default accounts ('demo/demo', 'admin/admin') that
```
Comment (aminelouati) [inline src/docs/admin/ProActiveAdminGuide.adoc:1304]: file based -> File-based
('demo/demo', 'admin/admin') -> (i.e., 'demo/demo', 'admin/admin')

Comment (nebilbenmabrouk) [inline src/docs/admin/ProActiveAdminGuide.adoc:1304]: Suggest to remove:
- server
- user passwords

By default, ProActive is configured to use *file based* authentication and has some default accounts ('demo', 'admin') that ..

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 3]
```diff
@@ -1322,16 +1325,17 @@ pa.rm.authentication.loginMethod=RMLDAPLoginMethod
...
+pa.scheduler.core.authentication.loginMethod=SchedulerFileLoginMethod
 ----
 
+[[_file]]
+=== File authentication
 
-=== File
-
-By default, the ProActive Resource Manager stores users accounts, passwords, group memberships (user or admins), and tenant memberships (group agglomerate or organization), in two files.
+By default, the ProActive server stores users accounts, passwords, group memberships (user or admins), and tenant memberships (group agglomerate or organization), in two files.
```
Comment (nebilbenmabrouk) [inline src/docs/admin/ProActiveAdminGuide.adoc:1338]: By default, ProActive stores ..

**Review by tao-xinxiu — APPROVED** [review]

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 4]
```diff
@@ -1403,12 +1407,16 @@ expert-ds:experts
 user:users
 ----
 
-When defined via <<_ldap,LDAP>>, the name of the <<_configure_tenant_attribute,LDAP attribute>> which will contain tenant information must be configured.
+When defined as a <<_ldap,LDAP attribute>>, the name of the <<_configure_tenant_attribute,attribute>> which will contain tenant information must be configured.
+
+When the authentication method is set to <<_ldap_with_multiple_domains,LDAP with Multiple Domains>>, the tenant is automatically set to the current user _LDAP domain_.
 
-Tenants are used to isolate groups of users, where users belonging to one tenant will not see the activities launched by users belonging to other tenants.
+Tenants can be used to isolate groups of users, where users belonging to one tenant will not see the activities launched by users belonging to other tenants.
```
Comment (nebilbenmabrouk) [inline src/docs/admin/ProActiveAdminGuide.adoc:1414]: Tenants allow users to work in isolated silos, where users belonging to one tenant **cannot** not see the jobs of users belonging to other tenants.

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 5]
```diff
@@ -1540,13 +1548,98 @@ The LDAP configuration is defined in `PROACTIVE_HOME/config/authentication/ldap.
...
+
+The LDAP module uses ldap filters to search for users and groups.
+
+===== User filters
+
+The user filter is defined using the property `pa.ldap.user.filter`.
+The search is performed according to a configurable ldap subtree, controlled by the property `pa.ldap.userssubtree`.
+
+`pa.ldap.userssubtree` must be defined according to the base distinguished name of the ldap server.
+It can be further refined to restrict the search to a specific subtree where the users are located.
```
Comment (nebilbenmabrouk) [inline src/docs/admin/ProActiveAdminGuide.adoc:1573]: I deem it is very useful to provide examples of LDAP search commands that allow to check whether the values of the filters are correct.

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 6]
```diff
@@ -1619,7 +1773,49 @@ There are two configuration options to control this mechanism (present in the `l
...
 
-=== PAM
+[[_ldap_with_multiple_domains]]
+=== LDAP authentication with Multiple Domains
+
+The ProActive server can be configured to authenticate users according to *multiple LDAP domains*.
+
+Following is a step-by-step procedure to configure it.
+
+1. Copy the file `PROACTIVE_HOME\config\authentication\ldap.cfg` into multiple files, one for each domain.
```
Comment (nebilbenmabrouk) [inline src/docs/admin/ProActiveAdminGuide.adoc:1783]: Copy the file `PROACTIVE_HOME\config\authentication\ldap.cfg` **multiple times in the same directory, such as to have one file copy per domain.**

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 7]
```diff
@@ -1619,7 +1773,49 @@ There are two configuration options to control this mechanism (present in the `l
...
+pa.rm.multi.ldap.config=domain1:config/authentication/ldap-domain1.cfg,domain2:config/authentication/ldap-domain2.cfg
+----
++
+5. In `config\scheduler\settings.ini`, uncomment the property `pa.scheduler.multi.ldap.config` and define it according to the ldap configuration files you created and each domain name:
++
+----
+pa.scheduler.multi.ldap.config=domain1:config/authentication/ldap-domain1.cfg,domain2:config/authentication/ldap-domain2.cfg
+----
+
+Once all the steps above are completed, the ProActive server can be started with a multi-domain ldap configuration.
```
Comment (nebilbenmabrouk) [inline src/docs/admin/ProActiveAdminGuide.adoc:1808]: Once all the steps above are completed, the ProActive server can be (re)started  **in order to make the** multi-domain ldap configuration **effective**.

**Review by nebilbenmabrouk — APPROVED** [review]
Approved with minor comments

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 8]
```diff
@@ -1291,15 +1291,17 @@ To be able to edit the default logo displayed in ProActive portals, you have to
 
 == User Authentication
 
-In order to use ProActive Scheduler every user must have an account. It supports three methods for authentication:
+In order to log into ProActive server, every user must have a registered account.
+ProActive Scheduler and Resource Manager support several methods to authenticate users:
```
[the remark this answers is not in the document]
Comment (fviale) [reply src/docs/admin/ProActiveAdminGuide.adoc:1295]: ok

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 9]
```diff
@@ -1619,7 +1773,49 @@ There are two configuration options to control this mechanism (present in the `l
...
+config\authentication\ldap-domain1.cfg
+config\authentication\ldap-domain2.cfg
+----
++
+2. Using the procedure described in chapter <<_ldap>>, configure each ldap file according to the domain used (each domain will have its own ldap server, bind login name, user subtree, etc.).
++
+3. Follow the steps described in chapter <<_select_authentication_method>> to configure `RMMultiLDAPLoginMethod` and `SchedulerMultiLDAPLoginMethod`.
++
+4. In `config\rm\settings.ini`, uncomment the property `pa.rm.multi.ldap.config` and define it according to the ldap configuration files you created and each domain name:
++
```
Comment (aminelouati) [inline src/docs/admin/ProActiveAdminGuide.adoc:1797]: to the ldap configuration files you created and each domain name -> to the ldap configuration files you created and their corresponding domain names:

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 10]
```diff
@@ -1619,7 +1773,49 @@ There are two configuration options to control this mechanism (present in the `l
...
+----
+pa.rm.multi.ldap.config=domain1:config/authentication/ldap-domain1.cfg,domain2:config/authentication/ldap-domain2.cfg
+----
++
+5. In `config\scheduler\settings.ini`, uncomment the property `pa.scheduler.multi.ldap.config` and define it according to the ldap configuration files you created and each domain name:
++
```
Comment (aminelouati) [inline src/docs/admin/ProActiveAdminGuide.adoc:1803]: same as above

**Review by aminelouati — APPROVED** [review]

**Pull request merged**